kotnetcli/communicator/plaintextc.py

```diff
@@ -86,129 +77,69 @@ def print_generic_balk(self, percentage, style, color, stop_color, stop_style):
         color + percentagestring + "%" + \
         stop_color + "]" + stop_style
 
-    ## Encapsulates the printing of a "balk" string on stdout
-    ## Override this method to change the appearance of the printed string.
-    def print_balk(self, percentage):
-        self.print_generic_balk(percentage, "", "", "", "")
+    def print_bar(self, percentage):
+        self.print_generic_bar(percentage, "", "", "", "")
 
-    #### 2. communicator method implementations common for both login and logout ####
+    def finalize_session(self, success):
+        cursor.show()
 
-    #def eventPingFailure(self):
-    #    self.printerr("Niet verbonden met het KU Leuven-netwerk.")
-    #    
-    #def eventPingAlreadyOnline(self):
-    #    self.printerr("U bent al online.")
+    ################## COMMON LOGIN/LOGOUT COMMUNICATOR INTERFACE ##################
 
-    def eventKotnetVerbindingStart(self):
-        self.print_wait("Kotnetverbinding testen.... ")
-        
-    def eventKotnetVerbindingSuccess(self):
+    def eventCheckNetworkConnection(self):
+        self.print_txt("Kotnetverbinding testen.... ")
+        self.print_wait()
+    
+    def eventGetData(self):
         self.print_success()
+        self.print_txt("Gegevens ophalen........... ")
+        self.print_wait()
     
-    def eventKotnetVerbindingFailure(self):
-        self.print_fail()
-
-    def eventNetloginSuccess(self):
+    def eventPostData(self):
         self.print_success()
-    def eventNetloginFailure(self):
-        self.print_fail()
+        self.print_txt("Gegevens opsturen.......... ")
+        self.print_wait()
         
-    def eventKuleuvenStart(self):
-        self.print_wait("KU Leuven kiezen........... ")
-    def eventKuleuvenSuccess(self):
+    def eventProcessData(self):
         self.print_success()
-    def eventKuleuvenFailure(self):
-        self.print_fail()
+        self.print_txt("Gegevens verwerken......... ")
+        self.print_wait()
 
-    def eventInvoerenStart(self):
-        self.print_wait("Gegevens invoeren.......... ")
-    def eventInvoerenSuccess(self):
-        self.print_success()
-    def eventInvoerenFailure(self):
+    def eventFailure(self, code):
         self.print_fail()
+        ## fail gracefully
+        self.finalize_session(False)
+        ## provide the user with an appropriate error message
```

Comment:
this actually smells very bad. I'm working on refactoring this to use explicit communicator methods for each kind of failure